.github/workflows/workflow.yml

```diff
@@ -45,12 +48,42 @@ jobs:
             exit 1
           fi
 
+      - name: Generate date-based cache key
+        id: key
+        run: |
+          # Put year and week number in steps.key.outputs.week
+          # This command "works" for both PowerShell and POSIX shell
+          echo ":$(Write-Output ':')set-output name=week::$(Get-Date -UFormat '%Y-%W')-"
+          echo ":`echo :`set-output name=week::`date -u '+%Y-%W'`-"
```

Comment:
This is possibly my most egregious hack of the last decade 😉 The aim is to have a single step but which runs in both PowerShell and `sh`.